test/CMakeLists.txt

```diff
@@ -25,7 +25,16 @@ if(BUILD_TESTS)
 
   add_executable(UnitTests ${TEST_SOURCES})
   target_include_directories(UnitTests PRIVATE /opt/rocm ${GTEST_INCLUDE_DIRS})
-  target_link_libraries(UnitTests PRIVATE ${GTEST_BOTH_LIBRARIES}  PRIVATE rccl)
+  target_link_libraries(UnitTests PRIVATE ${GTEST_BOTH_LIBRARIES})
+
+  # UnitTests using static library of rccl requires passing rccl
+  # through -l and -L instead of command line input.
+  if(BUILD_STATIC)
+    add_dependencies(UnitTests rccl)
+    target_link_libraries(UnitTests PRIVATE dl rt numa -lrccl -L${CMAKE_BINARY_DIR})
```

Comment:
Does rccl itself depend on these libraries? If so, shouldn't they be specified through target_link_libraries(rccl INTERFACE ...) ?

Reply:
Right, RCCL depends on these libraries, but normally it should have been brought in when we set link to hip::device (which includes them from either hsa-runtime::hsa-runtime64 or other deps). But I couldn't figure out how to force CMake to resolve something like hip::device into its list of link deps. 

Also, we may need to support librccl.a as an input file instead of using -L and -l?

Reply:
It looks like I need to explicitly put the static libraries into the PRIVATE libraries for UnitTests. I'm not sure how to work around it, but it doesn't seem like its picking up all the dependencies in static mode. I checked hip::device, and the dependency on hsa-runtime64::hsa-runtime64 is a PRIVATE one.